File: Arrays & Strings/1_uncompress.java
class Source {
    public static String uncompress(String s) {
        // two pointers approach
        String result = "";
        int startIndex = 0;
        int currentIndex = 0;

        while (currentIndex < s.length()) {
            if (Character.isDigit(s.charAt(currentIndex))) {
                // is digit so increase window
                currentIndex += 1;
            } else {
                // char found -> process window
                int numberChar = Integer.parseInt(s.substring(startIndex, currentIndex));
                for (int i = 0; i < numberChar; i++) {
                    result += s.charAt(currentIndex);
                }
                // reset two pointers
                currentIndex += 1;
                startIndex = currentIndex;
            }
        }

        return result;
    }
}

File: 1. Arrays & Strings/1_uncompress/1_uncompress.py
def uncompress(s):
    # two pointers approach
    result = ""
    start_index = 0
    current_index = 0

    while current_index < len(s):
        if s[current_index].isdigit():
            # is digit so increase window
            current_index += 1
        else:
            # char found -> process window
            number_char = int(s[start_index:current_index])
            for _ in range(number_char):
                result += s[current_index]
            # reset two pointers
            current_index += 1
            start_index = current_index

    return result
